### PR TITLE
Initial support for various Application Insights resources

### LINKED ISSUE #1305: Feature: Support Application Insights availability tests, alert rules
We maintain ARM templates for Application Insights availability tests and alert rules, which are deployed on top of an application's Application Insights deployment. It'd be nice to deploy these through ASO as well, so we don't have an extra set of application-specific infrastructure to manage that's not represented in Kubernetes.

### PR DESCRIPTION
**Pull request by matthchr**
Add the following resources:
* microsoft.insights/components
* microsoft.operationalinsights/workspace
* microsoft.insights/webtests

This partially resolves #1305.

In the process of adding these resources I ran into a number of issues:
* #1997
* #1998
* #1999

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/config/samples/insights/v1alpha1api20180501preview_webtest.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: insights.azure.com/v1alpha1api20180501preview
+kind: WebTest
+metadata:
+  name: samplewebtest
+  namespace: default
+spec:
+  tags:
+    "hidden-link:/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/aso-sample-rg/providers/microsoft.insights/components/sampleappinsights": Resource
```
Comment (babbageclunk) [inline v2/config/samples/insights/v1alpha1api20180501preview_webtest.yaml:8]: Weird, so the link between the test and the top-level component is by the test being tagged for the component?

File: v2/internal/controllers/crd_insights_component_test.go [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	exists, _, err := tc.AzureClient.HeadByID(
+		tc.Ctx,
+		armId,
+		string(insights.ComponentsSpecAPIVersion20200202))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(exists).To(BeFalse())
+}
+
+func Insights_WebTest_CRUD(tc *testcommon.KubePerTestContext, rg *resources.ResourceGroup, component *insights.Component) {
+	horribleHiddenLink := fmt.Sprintf("hidden-link:%s", to.String(component.Status.Id))
```
Comment (babbageclunk) [inline v2/internal/controllers/crd_insights_component_test.go:82]: Heh.

File: v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ import (
 // AddCrossResourceReferencesStageID is the unique identifier for this pipeline stage
 const AddCrossResourceReferencesStageID = "addCrossResourceReferences"
 
-var armIDDescriptionRegex = regexp.MustCompile("(?i)(.*/subscriptions/.*?/resourceGroups/.*|ARM ID|Resource ID)")
+var armIDDescriptionRegex = regexp.MustCompile("(?i)(.*/subscriptions/.*?/resourceGroups/.*|ARM ID|Resource ID|resourceId)")
```
Comment (babbageclunk) [inline v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go:24]: :heart:

Comment (babbageclunk) [reply v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go:24]: There might be another case that I should add to that regex, I saw some other missed references in the Redis resources...

**Review by babbageclunk — APPROVED** [review]
Looks great to me!

**Review by Porges — APPROVED** [review]

File: v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go (hunk 3 above)
Comment (matthchr) [reply v2/tools/generator/internal/codegen/pipeline/add_cross_resource_references.go:24]: Yeah unfortunately it's hard to be 100% perfect here. If you've seen other cases/examples definitely feel free to add them

**Pull request merged**